### ISSUE TITLE
Missing FileType in error string

### ISSUE THREAD

**Issue by greyscaled** [GlareDB/glaredb#2405]
I added a file in GCS as an external table. I am trying to query it, and the error string has a blank after "FileType":

> Error during planning: Unable to fetch table provider for 'public.gcs_test': This feature is not implemented: Unknown FileType:

Unfortunately, I don't have details (currently) on what is in that file or what type of file it is, but I feel like the error string could be more helpful regardless.

**Comment by hemanth94** [comment]
@greyscaled I was trying to understand this.
I'm assuming your query didn't have any options about "format" and the file didn't have any extension. Both at the same time.

It seems(for me) we cannot add additional details in error message as it occurs in datafusion-common crate.
Please correct me if I'm missing something. Thanks

<img width="571" alt="Screenshot 2024-01-12 235817" src="https://github.com/GlareDB/glaredb/assets/17337455/b903be8f-b978-4cc1-aa29-307c5dd7dcdf">



The code receives an empty string as filetype 
<img width="359" alt="Screenshot 2024-01-13 000205" src="https://github.com/GlareDB/glaredb/assets/17337455/2b44d5c8-a137-47e7-87b8-9be89061604c">

**Comment by greyscaled** [comment]
> @greyscaled I was trying to understand this. I'm assuming your query didn't have any options about "format" and the file didn't have any extension. Both at the same time.

For context, I am running these queries in [GlareDB Cloud](https://console.glaredb.com)

There, I have a deployment for which I've added an [`EXTERNAL TABLE`](https://docs.glaredb.com/docs/data-sources/supported/gcs.html#gcs-file-as-an-external-table) a long time ago. I was trying to query it:

```sql
SELECT * FROM public.gcs_test LIMIT 10;
```

In this regard, there is no options about `format`, which I believe is more pertinent when using file inference:

```sql
SELECT * FROM './path/to/my/file.csv'
```

or using table functions like [`csv_scan`](https://docs.glaredb.com/glaredb/sql-functions/csv_scan/).

I don't know what was originally in the file, and if it even exists anymore, which I've outlined in this issue: https://github.com/GlareDB/glaredb/issues/2406 . To be clear, because I don't remember where that file exists (which bucket or file name) AND glaredb doesn't expose that metadata to me, I can't really verify.

> It seems(for me) we cannot add additional details in error as it occurs in datafusion-common crate.

With all this context in mind, I'll lend the following thoughts:

- When a data source is added to glaredb (in this case `CREATE EXTERNAL TABLE`), I'd want to be able to look back on this details (which is what #2406 is about, not this ticket).

- When a data source is added in this manner, the path to the file is set. If the file no longer exists or moves or changes extension, I'd expect a more clear error message about that at a higher level, rather than a lower level error message.

- When a data source is added, should its type be inferred and "set"? So for example, if I had a file without an extension containing CSV data, and `CREATE EXTERNAL TABLE`, should there be metadata somewhere saying this file type is `csv`? I don't know if that's expected per se, but is worth asking.

In terms of the DF crate error message, what I'd expect is that if it is the case that the file no longer exists, that we'd get a different error message. If it's the case that I had a file without an extension, and then changed the contents of the file, I'm unsure what to expect but would want to make sure that behavior is defined.
